### PR TITLE
[Feat] Allow API Namespace to auto clean resources

### PR DESCRIPTION
**Pull request by Pralish**
Addresses: https://github.com/restarone/violet_rails/issues/1383

https://user-images.githubusercontent.com/50227291/227561742-94afc461-9731-49e5-a759-06492b8b84fd.mov

### CODE REVIEW

File: app/controllers/comfy/admin/api_namespaces_controller.rb [hunk 1]
```diff
@@ -84,21 +85,7 @@ def create
 
   # PATCH/PUT /api_namespaces/1 or /api_namespaces/1.json
   def update
-    respond_to do |format|
```
Comment (donrestarone) [inline app/controllers/comfy/admin/api_namespaces_controller.rb:87]: @Pralish I appreciate the code DRY'ing

**Comment by donrestarone** [comment]
@Pralish since we are changing the view on the API Namespace #show page, lets take this opportunity to move the tab bar to the top:
Screen Shot 2023-03-24 at 3 14 08 PM
Screen Shot 2023-03-24 at 3 13 57 PM

File: lib/tasks/maintenance.rake [hunk 2]
```diff
@@ -33,4 +33,21 @@ namespace :maintenance do
     end
     p "cleared old discarded api_actions @ #{Time.now}"
   end
+
+  task :purge_old_api_resources => [:environment] do 
+    Subdomain.all_with_public_schema.each do |subdomain|
+      Apartment::Tenant.switch subdomain.name do
+        ApiNamespace.where.not(purge_resources_older_than: ApiNamespace::RESOURCES_PURGE_INTERVAL_MAPPING[:never]).each do |api_namespace|
```
Comment (donrestarone) [inline lib/tasks/maintenance.rake:40]: lets read in batches, there is a huge number of records

https://api.rubyonrails.org/classes/ActiveRecord/Batches.html

**Comment by donrestarone** [comment]
🧪 @Pralish  when the clean up frequency is specified by submitting the form, a clean up should happen (unless they specify never, of course)

**Comment by donrestarone** [comment]
🧪 Test plan

1. when `allow settings` privilege is granted, the user is able to access the clean up frequency ⌛ 
2. when `allow settings` privilege is not granted, the user is not  able to access the clean up frequency ⌛ 
3. when the clean up frequency is specified by submitting the form, a clean up should happen ❌ 
4. clean up is performed on eligible resources only ⌛

**Comment by Pralish** [comment]
> @Pralish since we are changing the view on the API Namespace #show page, lets take this opportunity to move the tab bar to the top: Screen Shot 2023-03-24 at 3 14 08 PM Screen Shot 2023-03-24 at 3 13 57 PM

@donrestarone , Does that mean API resources list will be part of the tabs/tab-pane?

**Comment by donrestarone** [comment]
@Pralish no it won't be part of the tabs pane. 

The goal is to show the tab options and resource list without having the user scroll up and down needlessly.

**Comment by donrestarone** [comment]
@Pralish nice touch with the javascript confirmation before enqueueing the purge job!

LGTM ✅

File: lib/tasks/maintenance.rake (hunk 2 above)
Comment (donrestarone) [reply lib/tasks/maintenance.rake:40]: @Pralish awaiting this change before merging ⏫

**Comment by donrestarone** [comment]
after my comment is addressed about reading in batches when purging, we can merge

File: lib/tasks/maintenance.rake (hunk 2 above)
Comment (Pralish) [reply lib/tasks/maintenance.rake:40]: @donrestarone , This line reads the api_namespaces for a single subdomain. 

Here's the method that will read and destroy api_resources in batches.
https://github.com/restarone/violet_rails/pull/1492/files#diff-03241a9bad7359398fa0da5610d4fd762f027297ee8acb1326c900d1d8fe49f1R397-R399

Comment (donrestarone) [reply lib/tasks/maintenance.rake:40]: yes, but we should use the same method for performing the destruction in batches. 

The problem right now is that we delete in 2 different ways. The rake task can overload a system from the background if it reads too much data.

Comment (donrestarone) [reply lib/tasks/maintenance.rake:40]: also, we should read in batches here `ApiNamespace.where.not(purge_resources_older_than: ApiNamespace::RESOURCES_PURGE_INTERVAL_MAPPING[:never]).each`

Comment (donrestarone) [reply lib/tasks/maintenance.rake:40]: actually @Pralish Im seeing it now-- we don't have to read the namespaces in batches. This is fine ✅

**Pull request merged**